### PR TITLE
Plane: rework stickmixing to use per mode functions

### PR DESCRIPTION
**Pull request by IamPete1**
Moves stick mixing to be a per mode method.

### CODE REVIEW

File: ArduPlane/mode_fbwa.cpp [hunk 1]
```diff
@@ -35,3 +35,14 @@ void ModeFBWA::update()
...
     }
 }
+
+// Return true if stick mixing of any type is enabled
+bool ModeFBWA::allows_stick_mixing() const
+{
+    if (plane.failsafe.rc_failsafe && plane.g.fs_action_short == FS_ACTION_SHORT_FBWA) {
+        // don't do stick mixing in FBWA glide mode
+        return false;
+    }
```
Comment (IamPete1) [inline ArduPlane/mode_fbwa.cpp:45]: This check has been narrowed in scope, we used to do it in all modes, now only in FBWA. I think this correct, but it is a change.

**Comment by IamPete1** [comment]
Costs 36 bytes on Durandal and 26 bytes on Pixhawk1-1M and MatekF405.

**Comment by tridge** [comment]
@IamPete1 I think we can remove the "stabilize" stick mixing now, I don't think anyone uses it

**Comment by IamPete1** [comment]
Now direct stick mixing is gone I suspect this is a bit more of a flash loss. But I think still a improvement in code structure.

**Comment by IamPete1** [comment]
This is much less of a win now, closing.

**Pull request closed without merging**